### ISSUE TITLE
Proposal: Form.Custom

### ISSUE THREAD

**Issue by astanciu** [auth0/cosmos#506]
It would be really nice to have a generic `Form.Custom` element, that preserves the spacing, sizing, label, etc... but allows for inserting any custom component(s). 

For ex, I'm testing out Cosmos trying to build this form in Phenix:
![image](https://user-images.githubusercontent.com/1675025/38754159-bb63fa7a-3f2e-11e8-8d88-171f8b3440a8.png)

As you can see, `Other Names`, `Type`, `Resource Owner`, `Meta Tags` are either custom components or UI composition outside the norm (such as Type). 

Would be great, even as a migration helper, if I could do something like:

```
<Form>
  <Form.Custom label="Other Names">
      <CustomTagInput ... />
      helper text, etc...
  </Form.Custom>
</Form>
```

Without this, I don't know if/how I can partially use Cosmos in a form like this w/o breaking the layout..

**Comment by landitus** [comment]
All of the components you mention are just missing from Cosmos, but will be in the first release. The one use case that's interesting for the Custom Field is `Resource Owner`, where there's a custom behavior to show hide elements (selects). So it's a interesting idea. 

That show/hide could be achieved with 2 fields (one without a label), but still. This Custom field could be useful while we build the rest of the library.

**Comment by siddharthkp** [comment]
First attempt: https://github.com/auth0/cosmos/pull/530

**Comment by nkohari** [comment]
Closed by #530. We can revisit this if we want to expand it in the future.
